File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03017/p03017_num0_parsed.py
def is_possible(N, A, B, C, D, S):
    # Check if any rock is blocking the path between A and C
    for i in range(A+1, C):
        if S[i-1] == '#' and S[i] == '#':
            return 'No'
    
    # Check if any rock is blocking the path between B and D
    for i in range(B+1, D):
        if S[i-1] == '#' and S[i] == '#':
            return 'No'
    
    # Check if A can reach C
    if C > D and C < B:
        return 'No'
    
    # Check if B can reach D
    if D > C and D < A:
        return 'No'
    
    return 'Yes'


# Read input
N, A, B, C, D = map(int, input().split())
S = input()

# Call the function and print the result
result = is_possible(N, A, B, C, D, S)
print(result)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03017/p03017_num0_parsed.cpp
#include <iostream>
#include <string>

bool isPossible(int N, int A, int B, int C, int D, std::string& S) {
    // Check if there is a rock between Snuke and Fnuke
    for (int i = B; i <= D; i++) {
        if (S[i - 1] == '#' && S[i] == '#') {
            return false;
        }
    }

    // Check if Snuke can reach C before Fnuke reaches D
    if (C < D) {
        for (int i = B; i <= D; i++) {
            if (S[i - 1] == '.' && S[i] == '.' && S[i + 1] == '.') {
                return true;
            }
        }
        return false;
    }

    // Check if Snuke can reach C and Fnuke can reach D simultaneously
    for (int i = A; i <= C; i++) {
        if (S[i - 1] == '.' && S[i] == '.' && S[i + 1] == '.') {
            return true;
        }
    }

    return false;
}

int main() {
    int N, A, B, C, D;
    std::string S;
    std::cin >> N >> A >> B >> C >> D >> S;

    if (isPossible(N, A, B, C, D, S)) {
        std::cout << "Yes" << std::endl;
    } else {
        std::cout << "No" << std::endl;
    }
}